exercises/grade-school/Tests/GradeSchoolTests/GradeSchoolTests.swift

```diff
@@ -3,80 +3,72 @@ import XCTest
 
 class GradeSchoolTests: XCTestCase {
 
-    // Workaround unit test liminitaiton on Optional Arrays/Sets
-    private func sameCollection<C: Collection>(_ result: C?, _ expected: [String]?) -> Bool where C.Iterator.Element == String {
-        guard let result = result, let expected = expected, similar(result, expected) else { return false }
-        for (index, student) in result.enumerated() {
-            guard index < expected.count && expected.contains(student) else { return false }
-        }
-        return true
-    }
-    private func similar<C: Collection>(_ result: C, _ expected: [String]) -> Bool {
-        return result.count.toIntMax() == IntMax(expected.count)
+    func sameArray(_ result: [String]?, _ expected: [String]? ) -> Bool {
+        guard let result = result, let expected = expected else { return false }
+        return result == expected
     }
 
-    func XCTAssertEqualCollection<C:Collection> (_ collectionS: C?, _ collectionA: [String]? ) where C.Iterator.Element == String {
-        XCTAssert(sameCollection(collectionS, collectionA))
+    func XCTAssertEqualArray (_ result: [String]?, _ expected: [String]? ) {
```

Comment:
I was doing that before but I needed to unwrap the value in cases like:
```XCTAssertEqual(result[3], expected[3])```

Because of that I used the helper functions to safely unwrap and do the assertion without having guards in every function that accesses the students in the grade.